pkg/engine/lifecycletest/pulumi_test.go

```diff
@@ -5053,3 +5053,103 @@ func TestDefaultParents(t *testing.T) {
 	assert.Equal(t, snap.Resources[0].URN, snap.Resources[1].Parent)
 	assert.Equal(t, snap.Resources[0].URN, snap.Resources[2].Parent)
 }
+
+func TestProviderDeterministicPreview(t *testing.T) {
```

Comment:
This is basically the same test I removed in https://github.com/pulumi/pulumi/pull/10203 but using randomSeeds for deterministic naming rather than sequence numbers.